src/test/java/org/openrewrite/java/testing/mockito/ReplacePowerMockitoIntegrationTest.java

```diff
@@ -661,4 +661,76 @@ void setUp() {
           )
         );
     }
+
+    @Test
+    void whenNew() {
+        //language=java
+        rewriteRun(
+          java(
+              """
+              import org.powermock.api.mockito.PowerMockito;
+              import static org.powermock.api.mockito.PowerMockito.*;
+
+              import org.junit.jupiter.api.Test;
+              import static org.junit.jupiter.api.Assertions.assertEquals;
+
+              public class MyTest {
+                  static class Generator {
+                      public int getLuckyNumber() {
+                        return 436;
+                      }
+                  }
+
+                  @Test
+                  public final void testNumbers() throws Exception {
+                      Generator mock2 = mock(Generator.class);
+                      PowerMockito.whenNew(Generator.class).withNoArguments().thenReturn(mock2);
+
+                      Generator gen = new Generator();
+                      when(gen.getLuckyNumber()).thenReturn(504);
+
+                      assertEquals(504, gen.getLuckyNumber());
+                  }
+
+                  public final String otherMethod() {
+                    return "no change here";
+                  }
+              }
+              """,
+              """
+              import static org.mockito.Mockito.when;
+              import static org.mockito.Mockito.mock;
+
+              import org.junit.jupiter.api.Test;
+              import org.mockito.MockedConstruction;
+
+              import static org.junit.jupiter.api.Assertions.assertEquals;
+
+              public class MyTest {
+                  static class Generator {
+                      public int getLuckyNumber() {
+                        return 436;
+                      }
+                  }
+
+                  @Test
+                  public final void testNumbers() throws Exception {
+                      try (MockedConstruction<Generator> mockGenerator = Mockito.mockConstruction(Generator.class)) {
+                          Generator mock2 = mock(Generator.class);
```

Comment:
Is this `Generator mock2 = mock(Generator.class);` line still relevant? It can probably just be removed as well?

It looks like you don't need it at all, if I just read some blog about it: https://www.baeldung.com/java-mockito-mockedconstruction#mocking-constructors.